src/router/task.js

```diff
@@ -0,0 +1,69 @@
+const express = require('express')
+const Task = require('../models/task')
+const router = new express.Router()
+
+router.post('/', async (req, res) => {
+    const todo = new Task(req.body)
+    todo.save().then(() => {
+        res.send(todo)
+    }).catch((e) => {
+        res.send(e)
+    })
+})
+
+router.get('/', async (req, res) => {
+    Task.find({}).then((users) => {
+        res.send(users)
+        }).catch((e) => {
+        res.status(400).send(e)
+    })
+})
+
+router.get('/:id', async (req, res) => {
+    try {
+        const task = await Task.findOne({ _id: req.params.id })
+
+        if (!task) {
+            return res.status(404).send()
+        }
+        res.send(task)
+    } catch (e) {
+        res.status(500).send()
+    }
+})
+
+router.patch('/:id', async (req, res) => {
+    const updates = Object.keys(req.body)
+    const allowedUpdates = ['title', 'description', 'completed']
+    const isValidOper = updates.every((update) => allowedUpdates.includes(update))
+
+    if (!isValidOper) {
```

Comment:
do we have proper exception handling? it's duplicated all over. `{error: string} `is used here. Another person will throw `{error: {details: "invalid updates}}`

Reply:
Task.findOneAndUpdate({_id: req.params.id}, req.body, {}, (err, doc) => {
        if (err) {
            res.status(404).send(err)
        }
        res.send(doc)
    })

Reply:
this should be in a single place, rather than a duplication in every method